test/e2e/Makefile

```diff
@@ -38,28 +38,43 @@ TOOLS_DIR := $(REPO_ROOT)/hack/tools
 BIN_DIR := bin
 TOOLS_BIN_DIR := $(TOOLS_DIR)/$(BIN_DIR)
 GINKGO := $(TOOLS_BIN_DIR)/ginkgo
+KUSTOMIZE := $(TOOLS_BIN_DIR)/kustomize
 
-.PHONY: ginkgo
-ginkgo:
+$(GINKGO): # Build ginkgo from tools folder.
 	cd $(TOOLS_DIR) && go build -tags=tools -o $(BIN_DIR)/ginkgo github.com/onsi/ginkgo/ginkgo
 
+$(KUSTOMIZE): # Build kustomize from tools folder.
+    cd $(TOOLS_DIR) && go build -tags=tools -o $(BIN_DIR)/kustomize sigs.k8s.io/kustomize/kustomize/v3
+
 ## --------------------------------------
-## Testing
+## Templates
 ## --------------------------------------
 
-TEST_E2E_DIR := $(REPO_ROOT)/test/e2e
+DOCKER_TEMPLATES := $(REPO_ROOT)/test/e2e/data/infrastructure-docker
```

Comment:
from @sedefsavas https://github.com/kubernetes-sigs/cluster-api/pull/3639#discussion_r489152007

> Can we put them under templates dir like other providers? kcp-adoption does not make much sense to put it under that though.
> If we will keep them under e2e/data then we can modify cluster-templates below to e2e-cluster-templates to be explicit about their purpose.

TBH, I find adding e2e redundant adding e2e to the targets in the `test/e2e/Makefile`